app/views/EditAry/index.tsx

```diff
@@ -1,24 +1,143 @@
+                id
+                assessmentId
+                title
+                leadGroup {
+                    id
+                    title
+                }
+                title
+                clientId
+                assignee {
+                    id
+                    displayName
+                    emailDisplay
+                }
+                publishedOn
+                text
+                url
+                attachment {
+                    id
+                    title
+                    mimeType
+                    file {
+                        url
+                    }
+                }
+                isAssessmentLead
+                sourceType
+                priority
+                confidentiality
+                status
+                source {
+                    ...OrganizationGeneralResponse
+                }
+                authors {
+                    ...OrganizationGeneralResponse
+                }
+                emmEntities {
+                    id
+                    name
+                }
+                emmTriggers {
+                    id
+                    emmKeyword
+                    emmRiskFactor
+                    count
+                }
+                entries {
+                    ...EntryResponse
+                }
+            }
+        }
+    }
+`;
+
+export type EntryImagesMap = { [key: string]: Entry['image'] | undefined };
+
 interface Props {
     className?: string;
 }
 
 function EditAry(props: Props) {
     const { className } = props;
 
-    const { leadId } = useParams<{ leadId: string }>();
+    const leadId = new URL(window.location.href).searchParams.get('source') ?? undefined;
     const { project } = useContext(ProjectContext);
 
-    const projectId = project?.id;
+    const projectId = project ? project.id : undefined;
+
+    const variables = useMemo(
+        (): LeadEntriesForAryQueryVariables | undefined => (
+            (leadId && projectId) ? { projectId, leadId } : undefined
+        ), [
+            leadId,
+            projectId,
+        ],
+    );
+
+    const [entryImagesMap, setEntryImagesMap] = useState<EntryImagesMap | undefined>();
+
+    const {
+        data: entriesForLead,
+    } = useQuery<LeadEntriesForAryQuery, LeadEntriesForAryQueryVariables>(
+        LEAD_ENTRIES_FOR_ARY,
+        {
+            skip: isNotDefined(variables),
+            variables,
+            onCompleted: (response) => {
+                const leadFromResponse = response?.project?.lead;
+                if (!leadFromResponse) {
+                    return;
+                }
+                const imagesMap = listToMap(
+                    leadFromResponse.entries
+                        ?.map((entry) => entry.image)
+                        .filter(isDefined),
+                    (d) => d.id,
+                    (d) => d,
+                );
+                setEntryImagesMap(imagesMap);
+            },
+        },
+    );
 
```

Comment:
Cases to handle:
1. New assessment is created with sourceId from URL
  * Check if assessment has already been created from this sourceId, if so redirect to appropriate assessment
2. Old assessment is being edited
  * Fetch all lead and entry details from provided assessmentId